selftests/functional/test_basic.py

```diff
@@ -476,6 +477,21 @@ def test_invalid_python(self):
         self.assertIn('1-%s:MyTest.test_my_name -> TestError' % test,
                       result.stdout)
 
+    @unittest.skipIf(process.system("which /bin/read", ignore_status=True),
+                     "/bin/read not available.")
+    def test_read(self):
+        try:
+            cmd = utils_path.find_command("read")
+        except utils_path.CmdNotFoundError:
+            self.skipTest("'read' command not available")
```

Comment:
@ldoktor sorry if I wasn't clear enough, but my point with the 2 skips was that the second one happens in the context of the test itself, something that at the very least, goes against what we preach about tests.  If you choose to the 2 skips, please make the second also a decorator.